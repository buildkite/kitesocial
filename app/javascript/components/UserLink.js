import * as React from 'react';

const UserLink = ({ user }) => (
  <a href={user.url}>
    @{user.name}
  </a>
);

export default UserLink;
