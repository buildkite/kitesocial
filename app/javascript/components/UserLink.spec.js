/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import UserLink from './UserLink';

it('renders as a link based on the user data', () => {
  const component = renderer.create(
    <UserLink
      user={{
        id: 1,
        name: "alice",
        url: "/users/1"
      }}
    />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
