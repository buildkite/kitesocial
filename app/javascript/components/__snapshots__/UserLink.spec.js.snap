// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as a link based on the user data 1`] = `
<a
  href="/users/1"
>
  @
  alice
</a>
`;
